backend/composer/services/graph_service.py

```diff
@@ -157,12 +157,12 @@ def compile_journey(connectivity_statement) -> dict:
     # Extract origins, vias, and destinations from the connectivity statement
     Via = apps.get_model('composer', 'Via')
     Destination = apps.get_model('composer', 'Destination')
+    Origin = apps.get_model('composer', 'AnatomicalEntity')
 
-    origins = list(connectivity_statement.origins.all())
-
-    vias = list(Via.objects.filter(connectivity_statement=connectivity_statement))
-    destinations = list(Destination.objects.filter(connectivity_statement=connectivity_statement))
-
+    vias = list(Via.objects.filter(connectivity_statement__id=connectivity_statement.id))
+    destinations = list(Destination.objects.filter(connectivity_statement__id=connectivity_statement.id))
+    origins = list(Origin.objects.filter(origins_relations__id=connectivity_statement.id).distinct())
```

Comment:
question: Why did we change from:
    origins = list(connectivity_statement.origins.all())
    to
     origins = list(Origin.objects.filter(origins_relations__id=connectivity_statement.id).distinct())

Reply:
Reason is here in the code - https://github.com/MetaCell/sckan-composer/blob/feature/SCKAN-324/backend/composer/services/graph_service.py#L24

It says:
```
Exception has occurred: AttributeError       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
'AnatomicalEntity' object has no attribute 'name'
  File "/home/dgk/metacell/sckan-composer/backend/composer/services/graph_service.py", line 24, in generate_paths
    paths.extend(create_paths_from_origin(origin, vias, destinations, [(str(origin.id), origin.name, 0)], destination_layer))
  File "/home/dgk/metacell/sckan-composer/backend/composer/services/graph_service.py", line 166, in compile_journey
    all_paths2 = generate_paths(origins, vias, destinations)
  File "/home/dgk/metacell/sckan-composer/backend/composer/migrations/0067_auto_20241218_0928.py", line 11, in update_journey_fields
    cs.journey_path = compile_journey(cs)
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/db/migrations/operations/special.py", line 193, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/db/migrations/migration.py", line 130, in apply
    operation.database_forwards(
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/db/migrations/executor.py", line 252, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards
    state = self.apply_migration(
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/db/migrations/executor.py", line 135, in migrate
    state = self._migrate_all_forwards(
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/core/management/commands/migrate.py", line 349, in handle
    post_migrate_state = executor.migrate(
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/core/management/base.py", line 96, in wrapped
    res = handle_func(*args, **kwargs)
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/core/management/base.py", line 448, in execute
    output = self.handle(*args, **options)
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/core/management/base.py", line 402, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/core/management/__init__.py", line 440, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/home/dgk/metacell/sckan-composer/backend/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/home/dgk/metacell/sckan-composer/backend/manage.py", line 22, in <module>
    main()
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/dgk/miniconda3/envs/composer/lib/python3.9/runpy.py", line 197, in _run_module_as_main (Current frame)
    return _run_code(code, main_globals, None,
AttributeError: 'AnatomicalEntity' object has no attribute 'name'
```

Reply:
I tried to access the name of an origin with the original code and I was able to:
![image](https://github.com/user-attachments/assets/01cd100a-2cb4-44e0-a968-5874909907d7)

Can you explain with a bit more detail in which conditions the error you shown occurs?

Reply:
@afonsobspinto did you do the check with the API call?

This happens only when we do the migration - possibly something with - `apps.get_model('composer', 'ConnectivityStatement')` in the migration file (ref - [0067 auto migration file](https://github.com/MetaCell/sckan-composer/blob/feature/SCKAN-324/backend/composer/migrations/0067_auto_20241218_0928.py#L8)). Also if you notice this is only for the `@properties` and not for the fields. 
So if you put a breakpoint to the above mentioned code, and then run the migration to db  - this should cause the error.